pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -474,6 +546,43 @@ func (p *PriorityQueue) flushUnschedulableQLeftover() {
 func (p *PriorityQueue) Pop() (*v1.Pod, error) {
 	p.lock.Lock()
 	defer p.lock.Unlock()
+	if util.EquivalenceClassEnabled() {
+		// This is a dead cycle. If activeQ is not empty, Pop() gets equivalenceClass from activeQ
+		// and returns the first pod, then ends the cycle. If activeQ is empty, Pop() waits atomically
+		// unlocks p.lock, suspends execution of the calling goroutine until new item is enqueued and
+		// will not result in deadlock problem. When Close() is called, the p.closed is set and the condition
+		// is broadcast, then cycle will end.
+		for {
+			for p.activeQ.Len() == 0 {
+				// When the queue is empty, invocation of Pop() is blocked until new item is enqueued.
+				// When Close() is called, the p.closed is set and the condition is broadcast,
+				// which causes this loop to continue and return from the Pop().
+				if p.closed {
+					return nil, fmt.Errorf(queueClosed)
+				}
+				p.cond.Wait()
+			}
+
+			obj, err := p.activeQ.Pop()
```

Comment:
If I Understand Correctly about your question. The first pod just like the flag pod. We get the equivalence class through the flag pod and iterate through `PodSet` to get the pods. If that first pod is schedulable, we can get the second pod in `PodSet` directly. Also like this:
```go
equivalenceClass.PodSet.Range(func(k, v interface{}) bool {
			// Because Range() accesses read-only snapshots, it's possible that the previous
			// pod's scheduling time is too long to cause data to expire, so choose
			// to get the current data directly through Load().
			pod, ok := equivalenceClass.PodSet.Load(k)
			if !ok {
				return true
			}

			// If the pod is scheduled to fail, doSchedule returns false and range stops
			// the iteration.
			if !sched.doSchedule(pod.(*v1.Pod)) {
				return false
			}
			equivalenceClass.PodSet.Delete(pod.(*v1.Pod).UID)
			return true
		})
```